lang/src/org/partiql/lang/typesystem/builtin/types/nonprimitive/CharType.kt

```diff
@@ -0,0 +1,45 @@
+package org.partiql.lang.typesystem.builtin.types.nonprimitive
+
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.ExprValueType
+import org.partiql.lang.typesystem.builtin.types.primitive.IntType
+import org.partiql.lang.typesystem.builtin.types.valueFactory
+import org.partiql.lang.typesystem.interfaces.type.BuiltInType
+import org.partiql.lang.typesystem.interfaces.type.ParametricType
+import org.partiql.lang.typesystem.interfaces.type.SqlType
+import org.partiql.lang.typesystem.interfaces.type.TypeParameters
+import org.partiql.lang.util.asIonInt
+
+/**
+ * The standard sql type CHARACTER(n), where "n" is the fixed number of characters of the string literal
+ */
+object CharType : BuiltInType(), ParametricType {
+    /**
+     * If the parameter's value is -1, it means the user did not explicitly specify the parameter of the VARCHAR type.
+     */
+    private const val unboundedLength = -1
+
+    override val typeAliases: List<String>
+        get() = listOf("char", "character")
+
+    override val exprValueType: ExprValueType
+        get() = ExprValueType.STRING
+
+    override val isPrimitiveType: Boolean
+        get() = false
+
+    override val requiredParameters: List<SqlType> = emptyList()
+
+    override val optionalParameters: List<Pair<SqlType, ExprValue>> = listOf(
+        IntType to valueFactory.newInt(unboundedLength)
+    )
+
+    override fun validateParameters(parameters: TypeParameters) {
+        // Check if length is larger than 0
+        val length = parameters[0]
+        val value = length.value.ionValue.asIonInt().longValue()
+        if (value <= 0) {
```

Comment:
Don’t we still need a `maxLength` for the when the user provides the `length`?

Reply:
I was thinking we could have a value less that `INTEGER.MAX_VALUE` to control the amount of text (the actual size in bytes) just like what postgres does with [1GB](https://www.postgresql.org/docs/current/datatype-character.html) value but that’s a non-blocker for now.